client/src/main/java/org/apache/falcon/ExtensionUtil.java

```diff
@@ -0,0 +1,223 @@
+                                           InputStream configStream) throws IOException, FalconException {
+        ClassLoader previousClassLoader = Thread.currentThread().getContextClassLoader();
+        Thread.currentThread().setContextClassLoader(extensionClassloader);
+
+        ServiceLoader<ExtensionBuilder> extensionBuilders = ServiceLoader.load(ExtensionBuilder.class);
+
+        List<Class<? extends ExtensionBuilder>> result = new ArrayList<>();
+
+        for (ExtensionBuilder extensionBuilder : extensionBuilders) {
+            result.add(extensionBuilder.getClass());
+        }
+
+        if (result.isEmpty()) {
+            throw new FalconException("Extension Implementation not found in the package of : " + extensionName);
+        } else if (result.size() > 1) {
+            throw new FalconException("Found more than one extension Implementation in the package of : "
+                    + extensionName);
+        }
+
+        ExtensionBuilder extensionBuilder = null;
+        try {
+            Class<ExtensionBuilder> clazz = (Class<ExtensionBuilder>) extensionClassloader
+                    .loadClass(result.get(0).getCanonicalName());
+            extensionBuilder = clazz.newInstance();
+        } catch (ClassNotFoundException e) {
+            throw new FalconCLIException("Extension Implementation class not found", e);
+        } catch (InstantiationException e) {
+            throw new FalconCLIException("Failed to instantiate extension implementation", e);
+        } catch (IllegalAccessException e) {
+            throw new FalconCLIException("Failed to instantiate extension implementation", e);
+        }
+
+        List<Entity> entities = extensionBuilder.getEntities(jobName, configStream);
+
+        Thread.currentThread().setContextClassLoader(previousClassLoader);
+        return entities;
+    }
+
+    public static List<Entity> loadAndPrepare(String extensionName, String jobName, InputStream configStream,
+                                              String extensionBuildLocation) throws IOException, FalconException {
+        Configuration conf = new Configuration();
+        FileSystem fs = FileSystem.get(conf);
+        String stagePath = createStagePath(extensionName, jobName);
+        List<URL> urls = ExtensionUtil.copyExtensionPackage(extensionBuildLocation, fs, stagePath);
+
+        List<Entity> entities = prepare(extensionName, jobName, configStream, urls);
+        ExtensionUtil.stageEntities(entities, stagePath);
+        return entities;
+    }
+
+    public static List<Entity> prepare(String extensionName, String jobName, InputStream configStream, List<URL> urls)
+        throws IOException, FalconException {
+        ClassLoader extensionClassLoader = ExtensionClassLoader.load(urls);
+        return ExtensionUtil.getEntities(extensionClassLoader, extensionName, jobName, configStream);
+    }
+
+    public static void stageEntities(List<Entity> entities, String stagePath)
```

Comment:
Even wondering what purpose this is serving, as we are not using the entities staged. Might only help in debugging, right? If so, please add comment to clearly state the intention.